docs/designs/transaction-design.md

```diff
@@ -0,0 +1,222 @@
+# Architecture Design Document for Transaction History Page
+
+## Introduction  
+
+### Purpose  
+This document outlines the architecture for implementing a Transaction History feature in the Plymouth House inventory system.
+It details the design of 
+- The database schema for tracking and logging transaction activities.
+- The API endpoints for CRUD operations on transaction data.
+- The front-end logic for displaying transaction history to admin users only.
+
+### Scope  
+The architecture includes:  
+- Database schema design and stored procedure for transaction history.  
+- API endpoints for recording and retrieving transactions data.
+- Front-end components for displaying transaction history to admins.  
+
+Out of scope:  
+- Authentication and authorization mechanisms.  
+- Detailed UI/UX design.
+
+## System Overview  
+The Transaction History system consists of:  
+- A SQL Server table for storing transaction records.
+- API endpoints for CRUD transaction data.
+- A front-end page for displaying transaction history to admin users.
+
+## Architecture Goals and Constraints
+
+### Goals
+1. Ensure transaction history is recorded with:
+    - Primary key for each transaction.
+    - User ID.
+    - Transaction GUID.
+    - Item ID.
+    - Transaction type (e.g., Add, Checkout).
+    - Total amount.
+    - Transaction date.
+2. Provide API endpoints for recording and retrieving transaction data only for admin users.
+
+### Constraints
+- Admin role management will control access to the transaction history data.
+- Transaction data must be logged in real-time during any item addition or checkout.
+- The transaction history page should be able to display all transactions for a specific user.
+- The transaction history page should be able to display detailed information for each transaction.
+- The transaction history page should be able to filter and search transactions based on various criteria.
+
+## Detailed Design  
+
+### Database Schema
+
+#### Database  
+
+**Tables:**  
+- **TransactionHistory**:  
+  - `TransactionID` (Primary Key) - Unique identifier for each transaction record (UUID or GUID).
+  - `UserID` (Foreign Key) - References the `Users` table to identify the admin user associated with the transaction.
```

Comment:
```suggestion
  - `UserID` (Foreign Key) - References the `Users` table to identify the user (admin or volunteer) associated with the transaction.
```

Reply:
This leads to a good observation. We don't have the capability to track the admin email for transactions. We should. [#196](https://das-ph-inventory-tracker.atlassian.net/browse/PIT-196)